core/src/core_tests/workflow_tasks.rs

```diff
@@ -1694,16 +1711,28 @@ async fn evict_missing_wf_during_poll_doesnt_eat_permit() {
                 archived: false,
             })
         });
-    let mut mock = MocksHolder::from_client_with_responses(mock, tasks, []);
+    let mut mock = single_hist_mock_sg(wfid, t, [ResponseType::OneTask(2)], mock, true);
     mock.worker_cfg(|wc| {
         wc.max_cached_workflows = 1;
         wc.max_outstanding_workflow_tasks = 1;
     });
     let core = mock_worker(mock);
 
-    // Should error because mock is out of work
-    assert_matches!(core.poll_workflow_activation().await, Err(_));
-    assert_eq!(core.available_wft_permits(), 1);
+    // We will get an eviction for the workflow (which lang never saw in the first place, but,
+    // that's fine) since it had an error while being created, and then we will evict it and
+    // the permit shall be restored.
```

Comment:
I verified, this should be okay in TS but we should probably test this flow in all Core based SDKs